src/metabase/driver/postgres.clj

```diff
@@ -752,13 +752,14 @@
 (defmethod driver/upload-type->database-type :postgres
   [_driver upload-type]
   (case upload-type
-    ::upload/varchar_255 "VARCHAR(255)"
-    ::upload/text        "TEXT"
-    ::upload/int         "BIGINT"
-    ::upload/float       "FLOAT"
-    ::upload/boolean     "BOOLEAN"
-    ::upload/date        "DATE"
-    ::upload/datetime    "TIMESTAMP"))
+    ::upload/varchar_255 [:varchar 255]
+    ::upload/text        :text
+    ::upload/int         :bigint
+    ::upload/pk          :serial ;; bit of a hack, but we need to get it in the right order with :primary-key
```

Comment:
This is a serious code smell. This function is not just returning database-types anymore. Plus how are we supposed to know the column is auto-incrementing and not just a non auto-incrementing PK?